src/server/indices/mixin.js

```diff
@@ -0,0 +1,39 @@
+import { IndicesService } from './service';
+import { searchIndicesRoute } from './routes';
+
+export function indicesMixin(kbnServer, server) {
+  const pre = {
+    /**
+    *  Create an instance of the `indices` service against the data cluster
+    *  @type {Hapi.Pre}
+    */
+    getIndicesDataService: {
+      assign: 'dataIndices',
+      method(req, reply) {
+        const dataCluster = req.server.plugins.elasticsearch.getCluster('data');
+        const callDataCluster = (...args) => (
+          dataCluster.callWithRequest(req, ...args)
+        );
+
+        reply(new IndicesService(callDataCluster));
+      }
+    },
+    /**
+    *  Create an instance of the `indices` service against the admin cluster
+    *  @type {Hapi.Pre}
+    */
+    getIndicesAdminService: {
+      assign: 'adminIndices',
+      method(req, reply) {
+        const dataCluster = req.server.plugins.elasticsearch.getCluster('admin');
+        const callDataCluster = (...args) => (
+          dataCluster.callWithRequest(req, ...args)
+        );
+
+        reply(new IndicesService(callDataCluster));
```

Comment:
I think it's an abstraction on top of it. The actual [route is configured here](https://github.com/chrisronline/kibana/blob/37962f43975392d6c9eed73cf77ea09fb76acb88/src/server/indices/mixin.js#L38) where you see [the hapi usage](https://github.com/chrisronline/kibana/blob/37962f43975392d6c9eed73cf77ea09fb76acb88/src/server/indices/routes/search_indices_route.js#L27). As a reference, I just did what [index patterns did](https://github.com/elastic/kibana/blob/master/src/server/index_patterns/mixin.js#L22)